warehouse/accounts/forms.py

```diff
@@ -279,20 +277,27 @@ def __init__(self, *args, request, **kwargs):
     def validate_email(self, field):
         # Additional checks for the validity of the address
         try:
-            address = Address(addr_spec=field.data)
-        except (ValueError, HeaderParseError):
+            resp = email_validator.validate_email(field.data, check_deliverability=True)
+        except email_validator.EmailSyntaxError as e:
             raise wtforms.validators.ValidationError(
                 self.request._("The email address isn't valid. Try again.")
-            )
+            ) from e
 
         # Check if the domain is valid
         extractor = TLDExtract(suffix_list_urls=())  # Updated during image build
-        domain = extractor(address.domain.lower()).registered_domain
+        domain = extractor(resp.domain.lower()).registered_domain
+        mx_domains = set()
+        if hasattr(resp, "mx") and resp.mx:
+            mx_domains = {
+                extractor(mx_host.lower()).registered_domain
+                for _prio, mx_host in resp.mx
+            }
+        all_domains = mx_domains.union({domain})
 
         if (
-            domain in disposable_email_domains.blocklist
+            any(d in disposable_email_domains.blocklist for d in all_domains)
             or self.request.db.query(
-                exists().where(ProhibitedEmailDomain.domain == domain)
+                exists().where(ProhibitedEmailDomain.domain.in_(all_domains))
```

Comment:
From conversation, @di suggests a separate ProhibitedMxDomains table and use that explicitly, or extend the existing domain with an `mx = true` column, so we could decide how widely we want to prohibit a given domain name.